lib/src/model/study/study.dart

```diff
@@ -1,11 +1,115 @@
+import 'package:collection/collection.dart';
+import 'package:dartchess/dartchess.dart';
 import 'package:fast_immutable_collections/fast_immutable_collections.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
 import 'package:lichess_mobile/src/model/common/id.dart';
 import 'package:lichess_mobile/src/model/user/user.dart';
 
 part 'study.freezed.dart';
 part 'study.g.dart';
 
+@Freezed(fromJson: true)
+class Study with _$Study {
+  const Study._();
+
+  const factory Study({
+    required StudyId id,
+    required String name,
+    required ({StudyChapterId chapterId, String path}) position,
+    required bool liked,
+    required int likes,
+    required UserId? ownerId,
+    required IList<String> topics,
+    required IList<StudyChapterMeta> chapters,
+    required StudyChapter chapter,
+  }) = _Study;
+
+  StudyChapterMeta get currentChapterMeta =>
+      chapters.firstWhere((c) => c.id == chapter.id);
+
+  factory Study.fromJson(Map<String, Object?> json) => _$StudyFromJson(json);
+}
+
+@Freezed(fromJson: true)
+class StudyFeatures with _$StudyFeatures {
+  const StudyFeatures._();
+
+  const factory StudyFeatures({
+    @JsonKey(defaultValue: false) required bool cloneable,
+    @JsonKey(defaultValue: false) required bool chat,
+    @JsonKey(defaultValue: false) required bool sticky,
+  }) = _StudyFeatures;
+
+  factory StudyFeatures.fromJson(Map<String, Object?> json) =>
+      _$StudyFeaturesFromJson(json);
+}
+
+@Freezed(fromJson: true)
+class StudyChapter with _$StudyChapter {
+  const StudyChapter._();
+
+  const factory StudyChapter({
+    required StudyChapterId id,
+    required StudyChapterSetup setup,
+    @JsonKey(defaultValue: false) required bool practise,
+    required int? conceal,
+    @JsonKey(defaultValue: false) required bool gamebook,
+    required StudyChapterFeatures features,
+  }) = _StudyChapter;
+
+  factory StudyChapter.fromJson(Map<String, Object?> json) =>
+      _$StudyChapterFromJson(json);
+}
+
+@Freezed(fromJson: true)
+class StudyChapterFeatures with _$StudyChapterFeatures {
```

Comment:
Done, I had to use `bool?` instead of `bool` then though, since the JSON response seems to not always actually contain these fields (and with a simple record we can't use `@Default`

Reply:
ah, solved this via `@JsonKey(fromJson: ...)`